src/Microsoft.Health.Fhir.Shared.Api/Features/Resources/Bundle/TransactionBundleValidator.cs

```diff
@@ -68,6 +65,12 @@ public async Task ValidateBundle(Hl7.Fhir.Model.Bundle bundle, CancellationToken
             }
         }
 
+        public async Task ResolveBundleReferences(EntryComponent entry, Dictionary<string, (string resourceId, string resourceType)> referenceIdDictionary, CancellationToken cancellationToken)
```

Comment:
Not sure if this would be the right location for this method. As per class name, this class should only handle validation of bundle?

Reply:
+1, removed from here